### ISSUE TITLE
[BUG] Access Violation error upon boot

### ISSUE THREAD

**Issue by gebdag** [vs-shirokii/gzdoom-rt#4]
### GZDoom version

The one shipped with 1.0.2

### Which game are you running with GZDoom?

Doom

### What Operating System are you using?

Windows 11

### Please describe your specific OS version

Windows 11 Pro

### Relevant hardware info

12700k, RTX 4090

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

The game gets to the Raytracing initialization step and then crashes with the following error message: "Code: C0000005 (Access Violation - tried to read address [0000000000000000](tel:0000000000000000))". 
Already tried to clear the config file. It worked fine yesterday. The only thing I installed in the mean time was the Wukong Benchmark off of Steam. Could it potentially be a dependency issue? 

### Steps to reproduce the behaviour.

Happens upon boot
Happens with both RTGL1 and without

### Provide a Log
[CrashReport.zip](https://github.com/user-attachments/files/16605362/CrashReport.zip)

**Comment by tioslash** [comment]
Same exact error here everytime I try to launch it, except I am using a Radeon 7900 XT gpu.

**Comment by kissandfly** [comment]
Same here. Windows 11 RTX4070

**Comment by gebdag** [comment]
I fixed the issue reinstalling the VC redistributable 
https://aka.ms/vs/17/release/vc_redist.x64.exe

**Comment by tioslash** [comment]
> I fixed the issue reinstalling the VC redistributable https://aka.ms/vs/17/release/vc_redist.x64.exe

Good to hear. Tried it here on my end and unfortunately did not work :/ Still error on launch.

**Comment by Sk-F** [comment]
Same here. Also reinstalled VC, didn't work. 

5800X3D + 6800XT

Mine says: 
Code: C0000005 (Access Violation - tried to read address 00000000000000008) 
Address: 00007ffa4a685c81

Couldn't even boot it, not even once.

**Comment by kissandfly** [comment]
Great works for me, thanks

**Comment by vvtunes** [comment]
Same error here, tried to reinstall VC redist to no effect.
v1.0.2, doom 2, 5800x3d + 7900xt, windows 11 24h2

Code: C0000005 (Access Violation - tried to read address 0000000000000008)
Address: 00007ffd9205a9b1

**Comment by sguart** [comment]
> I fixed the issue reinstalling the VC redistributable https://aka.ms/vs/17/release/vc_redist.x64.exe

it worked after i did a repair then reboot the system.

**Comment by Shageki** [comment]
Same for me:
v1.0.2, Doom 2 (Steam), Ryzen 5800X + 6800XT, Windows 11 Pro 23H2

Code: C0000005 (Access Violation - tried to read address 0000000000000008)
Address: 00007ffd9a955cc1

Repaired VC Distributable and rebooted, no joy.

**Comment by greasefart** [comment]
Same error: Code: C0000005 (Access Violation - tried to read address 0000000000000008)

5800X3D, 7900 XTX, latest stable AMD drivers, Windows 11 Pro, updated

- Reinstalled VC, rebooted, no luck
- Tried Steam Doom II, and DOOM2.wad from GOG distribution, no luck.
 
Can't wait to play! Looks awesome.

**Comment by vs-shirokii** [comment]
Eh, null pointer read somewhere.. can someone send a crash dump?

"Save report to disk" and send that file

**Comment by snoopy7548** [comment]
> Eh, null pointer read somewhere.. can someone send a crash dump?
> 
> "Save report to disk" and send that file

Here's my crash report with the same error.

[CrashReport.zip](https://github.com/user-attachments/files/16649116/CrashReport.zip)

**Comment by greasefart** [comment]
[CrashReport.zip](https://github.com/user-attachments/files/16649184/CrashReport.zip)

**Comment by tioslash** [comment]
Here´s mine:

[CrashReport.zip](https://github.com/user-attachments/files/16649297/CrashReport.zip)

**Comment by vs-shirokii** [comment]
Oof, something inside the AMD driver 🤔
I guess I misuse something without checking availability of some feature.

If someone is familiar with Vulkan SDK and Vulkan Configurator (to force Validation Layers, and output validation log), it would be helpful to understand where a problem might be

Low possibility, but maybe to update AMD drivers to latest until then..

**Comment by Gleniel** [comment]
> Oof, something inside the AMD driver 🤔 I guess I misuse something without checking availability of some feature.
> 
> If someone is familiar with Vulkan SDK and Vulkan Configurator (to force Validation Layers, and output validation log), it would be helpful to understand where a problem might be
> 
> Low possibility, but maybe to update AMD drivers to latest until then..

i think the fluid simulation is the issue, i found it on the comment on other issue.
https://github.com/vs-shirokii/gzdoom-rt/issues/14#issuecomment-2295427772

**Comment by tioslash** [comment]
> > Oof, something inside the AMD driver 🤔 I guess I misuse something without checking availability of some feature.
> > If someone is familiar with Vulkan SDK and Vulkan Configurator (to force Validation Layers, and output validation log), it would be helpful to understand where a problem might be
> > Low possibility, but maybe to update AMD drivers to latest until then..
> 
> i think the fluid simulation is the issue, i found it on the comment on other issue. [#14 (comment)](https://github.com/vs-shirokii/gzdoom-rt/issues/14#issuecomment-2295427772)

The gzdoom-rt2.ini file does not even show up for me in the /Documents/My Games/GZDoom/ folder after launching the mod once and getting the crash error unfortunately.

**Comment by vs-shirokii** [comment]
@tioslash 
You can try to start `gzdoom.exe` with `+rt_fluid 0` argument.

I.e. create .txt file around your .exe, put `gzdoom.exe +rt_fluid 0` in it. Rename .txt to .bat, and double-click that .bat file.

**Comment by Neo27484** [comment]
set your windows resolution to 1920x1080. This worked for me after the same error. Ingame is max 1080p possible

**Comment by tioslash** [comment]
> @tioslash You can try to start `gzdoom.exe` with `+rt_fluid 0` argument.
> 
> I.e. create .txt file around your .exe, put `gzdoom.exe +rt_fluid 0` in it. Rename .txt to .bat, and double-click that .bat file.

Hi, can confirm that this worked! I´ve created the .bat file and launched it without issues. Running 4k with FSR Quality mode and FG enabled on a 7900 XT. Latest drivers installed.

As soon as I enabled fluid simulation in the game options it crashed. So at least in my end this was definitely the issue.

**Comment by Gleniel** [comment]
> > @tioslash You can try to start `gzdoom.exe` with `+rt_fluid 0` argument.
> > I.e. create .txt file around your .exe, put `gzdoom.exe +rt_fluid 0` in it. Rename .txt to .bat, and double-click that .bat file.
> 
> Hi, can confirm that this worked! I´ve created the .bat file and launched it without issues. Running 4k with FSR Quality mode and FG enabled on a 7900 XT. Latest drivers installed.
> 
> As soon as I enabled fluid simulation in the game options it crashed. So at least in my end this was definitely the issue.

yeah, turning off the fluid simulation on ini file also worked for me, fluid simulation is definitely the issue causing it to crash on AMD GPUs, I'm using RX 6700 XT btw.

**Comment by greasefart** [comment]
`+rt_fluid 0` works for both 7900 XTX and 6800S. Anyone know what visuals I'm missing out on with this?

**Comment by kmonson** [comment]
> `+rt_fluid 0` works for both 7900 XTX and 6800S. Anyone know what visuals I'm missing out on with this?

Still seeing this with an 9070 XT in July 2025. As noted, turning off the fluid simulation allows it to run. Unfortunately this was not my computer (2080 SUPER, works great) demonstrating the issue, otherwise I'd take a crack at fixing it.
